### ISSUE TITLE
Nuxt usage

### ISSUE THREAD

**Issue by M-Barari** [talkhabi/vue-persian-datetime-picker#118]
سلام.
اگه میشه نحوه استفاده در nuxtjs رو هم به راهنما اضافه کنید.
برای من خطای document undifined میده!
مربوط به SSR هست؟

ممنون

**Comment by Hamidrezana** [comment]
try define component like this:
```
components: {
        datePicker: () => import('vue-persian-datetime-picker'),
}
```

**Comment by MNSY** [comment]
Except for use `mode: "client"`, if you dont need plugin globally you can import it as lazy import in your page/component:
`
  export default {
    name: "myComp",
    components:{
      DatePicker: () => import('vue-persian-datetime-picker'),
    },
}
`

**Comment by ghonchesefidi** [comment]
> First create plugin file in `plugins` directory and add this code there:
> 
> ```
> import Vue from "vue";
> 
> import VuePersianDatetimePicker from "vue-persian-datetime-picker";
> Vue.component("date-picker", VuePersianDatetimePicker);
> ```
> 
> then in `plugins` array in `nuxt.config.js` add this:
> 
> ```
> plugins: [
>   // ...others
>   { src: "~/plugins/datePicker", mode: "client" },
> ]
> ```

This worked for me on Nuxt. Thank you in advance.

**Comment by morteza-mortezai** [comment]
این کارا رو کردم  بازم مشکل حل نشد

**Comment by mohammadla75** [comment]
@M-Barari تونستی این مشکل رو برطرف بکنی؟

**Comment by M-Barari** [comment]
> @M-Barari تونستی این مشکل رو برطرف بکنی؟

```vue
<template>
  <client-only>
    <component :is="datePicker" />
  </client-only>
</template>

<script>
export default {
  computed:{
    datePicker(){
      if(process.client){
        return ()=>import('vue-persian-datetime-picker')
      }
    }
  }
}
</script>
```

**Comment by Hosein-NZF** [comment]
`  components: {
    [process.client && 'datePicker']: () => import('vue-persian-datetime-picker'),
  },`

**Comment by mes-mojahed** [comment]
I found it that if we use it like a single component :

// DataPicker.vue

<template>
    <client-only>
        <DatePicker v-model="date"></DatePicker>
    </client-only>
</template>

<script setup lang="ts">
import DatePicker from 'vue3-persian-datetime-picker'

const date = ref('')
</script>

we don't have any problem with it in nuxt 3 
I didn't test in nuxt 2

**Comment by Mohsen-graphyy** [comment]
Hi I got the same issue on Nuxt 3 SSR project, this bug comes from this package that does not support SSR, so you have two solutions:  
1. adding `.client` after your component name, like this: `BaseDatePicker.client.vue`.
3. can add `<ClientOnly> DatePicker <ClientOnly>` on root of the date picker component.

**Comment by mohammadla75** [comment]
i did it with this way, 
`components: {
    datePicker: () => import('vue-persian-datetime-picker'),
}`
@Mohsen-graphyy

**Comment by Mohsen-graphyy** [comment]
> i did it with this way, `components: { datePicker: () => import('vue-persian-datetime-picker'), }` @Mohsen-graphyy

yeah, I think this method can prevent extra bundle size on all of your project files that affect your application too. great

**Comment by okaroir** [comment]
سلام
خدا خیرتون بده یک نفر یه آموزش ضبط کنه ممنون میشم.

من از nuxt استفاده می کنم و از ساختار Composition API

راستی یه پکیج دیگه هم هست در آدرس زیر:
https://github.com/alireza-ab/vue3-persian-datepicker

**Comment by saeedhpro** [comment]
for use package in nuxtjs 3:
1. install vue3-persian-datetime-picker
2. create a client plugin: date-picker.client.ts
3. define plugin:
import Vue3PersianDatetimePicker from 'vue3-persian-datetime-picker'
export default defineNuxtPlugin(nuxtApp => {
    nuxtApp.vueApp.use(Vue3PersianDatetimePicker, {
        name: 'CustomDatePicker',
        props: {
            format: 'YYYY-MM-DD HH:mm',
            // other configs
        }
    })
})
4. use in any component like this:
<client-only>
    <CustomDatePicker></CustomDatePicker>
</client-only>
